src/snowflake/snowpark/dataframe.py

```diff
@@ -2245,14 +2256,30 @@ def natural_join(
             --------------------
             <BLANKLINE>
         """
-        join_type = kwargs.get("join_type") or how
+        join_type = create_join_type(kwargs.get("join_type") or how or "inner")
         join_plan = Join(
             self._plan,
             right._plan,
-            NaturalJoin(create_join_type(join_type or "inner")),
+            NaturalJoin(join_type),
             None,
             None,
         )
+
```

Comment:
somewhere here `self._ast` seems to be missing.  I.e., with
```
    def set_ast_ref(self, sp_dataframe_expr_builder: Any) -> None:
        """
        Given a field builder expression of the AST type SpDataframeExpr, points the builder to reference this dataframe.
        """
        # TODO: remove the None guard below once we generate the correct AST.
        if self._ast_id is not None:
            sp_dataframe_expr_builder.sp_dataframe_ref.id.bitfield1 = self._ast_id
```
I would expect to have the id of `self` being carried over to `ast` in some form. Put simply, intuitively the logic here should be something like
```
ast.lhs = self._ast
ast.rhs = other._ast
```
maybe I am missing something?

Reply:
Debugging with Arthur, we found that my test case included calls to `Dataframe.show()` because I lifted the tests directly from documentation examples.